### PR TITLE
Added stream() for #72

### PR DESCRIPTION
**Pull request by PaulWoitaschek**
* Added the stream method
* Deprecated old stream(barcode) and added migration helping in the documentation
* Added missing Nullable/NonNullable annotations
* Added missing Override annotations
* Fixed futures generic hickup

### CODE REVIEW

File: cache/src/main/java/com/nytimes/android/external/cache/Futures.java [hunk 1]
```diff
@@ -21,24 +21,24 @@ private Futures() {
     @Nullable
     public static <V> ListenableFuture<V> immediateFuture(@Nullable V value) {
         if (value == null) {
-            Futures.ImmediateSuccessfulFuture typedNull = Futures.ImmediateSuccessfulFuture.NULL;
-            return typedNull;
+            //noinspection unchecked safe because of erasure
+            return (ListenableFuture<V>) ImmediateSuccessfulFuture.NULL;
         } else {
-            return new Futures.ImmediateSuccessfulFuture(value);
+            return new Futures.ImmediateSuccessfulFuture<>(value);
```
Comment (digitalbuddha) [inline cache/src/main/java/com/nytimes/android/external/cache/Futures.java]: lol you're fixing guava code nice!

File: store/src/main/java/com/nytimes/android/external/store/base/impl/RealInternalStore.java [hunk 2]
```diff
@@ -227,6 +231,11 @@ void notifySubscribers(Parsed data) {
         return stream;
     }
 
+    @NonNull @Override
+    public Observable<Parsed> stream() {
+        return subject.asObservable();
```
Comment (digitalbuddha) [inline store/src/main/java/com/nytimes/android/external/store/base/impl/RealInternalStore.java]: lets add a `.share` before giving this out, should have been there all along 
http://blog.danlew.net/2016/06/13/multicasting-in-rxjava/

Comment (PaulWoitaschek) [inline store/src/main/java/com/nytimes/android/external/store/base/impl/RealInternalStore.java]: Why? It's a subject and it can have multiple subscribers.

Comment (digitalbuddha) [inline store/src/main/java/com/nytimes/android/external/store/base/impl/RealInternalStore.java]: Observable<Foo> fooObservable = store.stream();

barObservable= fooObservable.map(foo->foo.bar());

barObservable.subscribe()
barObservable.subscribe()

map gets called twice.  I'd prefer for it to be called once

Comment (digitalbuddha) [inline store/src/main/java/com/nytimes/android/external/store/base/impl/RealInternalStore.java]: actually forget what I said, this is different problem

**Pull request merged**